server/src/main/java/se/fortnox/reactivewizard/server/ConnectionCounter.java

```diff
@@ -7,8 +7,18 @@
 
 @Singleton
 public class ConnectionCounter {
-    private final AtomicLong connections = new AtomicLong(0);
-    private final Semaphore connectionsZero = new Semaphore(1);
+    private final AtomicLong connections;
+    private final Semaphore connectionsZero;
+
+    public ConnectionCounter() {
+        connections = new AtomicLong(0);
```

Comment:
Perhaps use the protected constructor to avoid divergation later?

Reply:
sounds like a good idea!